hoot-core/src/main/cpp/hoot/core/ops/FindIntersectionsOp.h

```diff
@@ -36,23 +37,57 @@
 namespace hoot
 {
 
+class FindIntersectionsVisitor;
 
 /**
  * Op. to find intersections
  */
-class FindHighwayIntersectionsOp : public OsmMapOperation
+class FindIntersectionsOp : public OsmMapOperation
 {
 public:
 
-  static std::string className() { return "hoot::FindHighwayIntersectionsOp"; }
+  static std::string className() { return "hoot::FindIntersectionsOp"; }
 
   static QString opsKey() { return "map.cleaner.transforms"; }
 
-  FindHighwayIntersectionsOp();
+  FindIntersectionsOp();
 
   virtual void apply(std::shared_ptr<OsmMap>& map) override;
 
-  virtual QString getDescription() const override { return "Identifies road intersections"; }
+  virtual QString getDescription() const override { return "Identifies generic intersections"; }
+
+  /**
+   *  Pure virtual function called by constructor to create the intersection visitor
+   */
+  virtual std::shared_ptr<FindIntersectionsVisitor> createVisitor() = 0;
```

Comment:
This pure virtual function allows the base class to do all of the work but do it on a different visitor.  Each derived operation class will create the associated visitor.